frontend/src/Pages/Favorites.tsx

```diff
@@ -0,0 +1,150 @@
+                    typedResponse.data.map((item) => ({
+                        id: item.id,
+                        content_id: item.content_id,
+                        name: item.name,
+                        type: item.type,
+                        thumbnail_url: item.thumbnail_url,
+                        description: item.description,
+                        visibility_status: item.visibility_status,
+                        open_content_provider_id: item.open_content_provider_id,
+                        content_url: item.content_url,
+                        channel_title: item.channel_title ?? '',
+                        provider_name: item.provider_name ?? ''
+                    }));
+                setFavorites(mappedFavorites);
+            } catch (err) {
+                setError('Error fetching favorites.');
+                console.error(err);
+            } finally {
+                setIsLoading(false);
+            }
+        };
+        void fetchFavorites();
+    }, [user]);
+
+    const handleUnfavorite = async (
+        contentId: number,
+        type: 'video' | 'library'
+    ) => {
+        try {
+            setIsLoading(true);
+            let endpoint = '';
+
+            if (type === 'video') {
+                endpoint = `videos/${contentId}/favorite`;
+            } else {
+                endpoint = `open-content/${contentId}/save`;
+            }
+
+            const favorite = favorites.find(
+                (fav) => fav.content_id === contentId && fav.type === type
+            );
+
+            if (!favorite) {
+                setError('Favorite not found.');
+                return;
+            }
+
+            const { id, name, open_content_provider_id } = favorite;
+            if (!name || !open_content_provider_id) {
+                setError('Favorite data is incomplete.');
+                return;
+            }
+
+            const payload = {
+                name,
+                content_id: contentId,
+                open_content_provider_id,
+                content_url: `/api/proxy/libraries/${contentId}/`
+            };
+
+            const response = await API.put(endpoint, payload);
+
+            if (response.success) {
+                await mutate('/open-content/favorites');
+                setFavorites((prevFavorites) =>
+                    prevFavorites.filter((fav) => fav.id !== id)
+                );
+
+                if (type === 'library') {
+                    await mutate('/api/libraries');
+                }
+            } else {
+                setError('Failed to unfavorite item.');
+            }
+        } catch (err) {
+            setError('Error unfavoriting item.');
+            console.error(err);
+        } finally {
+            setIsLoading(false);
+        }
+    };
+    if (isLoading) {
+        return <Loading />;
+    }
+
+    if (error && favorites.length !== 0) {
+        return <p className="text-red-500">{error}</p>;
```

Comment:
use text-error for this

Reply:
Actually, I would say that this probably shouldn't be rendered here. If there is an error, this would re render the whole contents of this page to show that error, when in reality we'd probably still want to show the user the favorites but instead just show them a toast telling them that we were unable to favorite/unfavorite their library. 
<img width="1512" alt="Screenshot 2024-11-22 at 10 17 07 AM" src="https://github.com/user-attachments/assets/c02f1a9f-3877-45e7-9b9a-41eaeee2e7e8">